File: bank-service/src/main/java/rs/edu/raf/model/ExchangeInvoice.java
package rs.edu.raf.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import rs.edu.raf.model.accounts.BankAccount;

import java.math.BigDecimal;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class ExchangeInvoice {

    @Id
    @Column(unique = true)
    @GeneratedValue(strategy = GenerationType.AUTO)
    @NotNull
    private Long id;
    @ManyToOne
    private BankAccount senderAccount;
    @ManyToOne
    private BankAccount receiverAccount;
    private BigDecimal senderAmount;
    @ManyToOne
    private Currency senderCurrency;
    @ManyToOne
    private Currency receiverCurrency;
    private BigDecimal exchangeRate;
    private BigDecimal profit;
    private Long dateAndTime;
}

File: bank-service-kotlin/src/main/kotlin/org/example/bankservicekotlin/model/ExchangeInvoice.kt
package org.example.bankservicekotlin.model

import jakarta.persistence.*
import org.example.bankservicekotlin.model.accounts.BankAccount
import java.math.BigDecimal

@Entity
class ExchangeInvoice(
    @Id @GeneratedValue(strategy = GenerationType.AUTO)
    var id: Long?,
    @ManyToOne
    val senderAccount: BankAccount,
    @ManyToOne
    val receiverAccount: BankAccount,
    val senderAmount: BigDecimal,
    @ManyToOne
    val senderCurrency: Currency,
    @ManyToOne
    val receiverCurrency: Currency,
    val exchangeRate: BigDecimal,
    val profit: BigDecimal,
    val dateAndTime: Long
) {
}